src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -69,7 +69,7 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsStandbyConfiguration = true;
             }
 
-            options.IsFileSystemReadOnly = IsZipDeployment(out bool isScmRunFromPackage);
+            options.IsFileSystemReadOnly |= IsZipDeployment(out bool isScmRunFromPackage);
```

Comment:
To test this fix, I had to simulate specialization locally. That requires configuration that ensures placeholder channels are used. For this specific change, the check [here](https://github.com/Azure/azure-functions-host/blob/b1c5107d2759c1a7a916fffda8d244ae346127b5/src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs#L152) requires a RO file system. So I'm making this change to facilitate testability, however it's a logical change to make regardless.